Makefile

```diff
@@ -2147,6 +2147,15 @@ GIT-USER-AGENT: FORCE
 		echo '$(GIT_USER_AGENT_SQ)' >GIT-USER-AGENT; \
 	fi
 
+GIT_BUILT_FROM_COMMIT = $(shell \
```

Comment:
very minor nit: using := rather than plain = would avoid multiple evals of the sub-shell expression.
not sure it is worth the bother though.

Reply:
Wouldn't `:=` _force_ the evaluation, even if the variable is not even used?

Reply:
I think both of you are right. In this case, maybe a deferred single expansion would work (won't evaluate until needed, and only evaluates once)? 

```
GIT_BUILT_FROM_COMMIT = $(eval GIT_BUILT_FROM_COMMIT := $$(shell \
	GIT_CEILING_DIRECTORIES="$$(CURDIR)/.." \
	git rev-parse -q --verify HEAD 2>/dev/null))$(GIT_BUILT_FROM_COMMIT)
```

`CURL_LDFLAGS` and `CURL_CFLAGS` do something similar ([context](https://lore.kernel.org/git/20200404145829.GB679473@coredump.intra.peff.net/)).

Reply:
But a `?=` would only evaluate unless set, right? So if you specify via the command-line, it won't be evaluated. And once evaluated, it is set, and won't evaluate again either.